cmd/capture.go

```diff
@@ -125,69 +136,96 @@ can then be used to converge locally.`,
+			requestGatherSources(repoDirName)
+			baseUserPath := requestCookbookPath(remainingCBs)
+			for len(remainingCBs) > 0 && baseUserPath != "" {
+				remainingCBs, err = resolveCookbooks(cookbookDirName, remainingCBs, repoDirName, baseUserPath)
 				if err != nil {
 					return err
 				}
-				if len(cookbooks) == 0 {
-					break
-				}
-				path = requestCookbookPath(cookbooks)
+				baseUserPath = requestCookbookPath(remainingCBs)
 			}
 
-			if len(cookbooks) > 0 {
-				fmt.Printf(CookbooksNotSourcedTxt, fmt.Sprintf("%s/%s", dirName, "cookbooks"),
-					formatCookbooks(cookbooks),
-				)
+			if len(remainingCBs) > 0 {
+				fmt.Printf(CookbooksNotSourcedTxt, fmt.Sprintf("%s/%s", repoDirName, cookbookDirName),
+					formatCookbooks(remainingCBs))
 			}
 
-			fmt.Print(CookbookCaptureCompleteTxt, dirName)
+			fmt.Print(CookbookCaptureCompleteTxt, repoDirName)
 			return nil
+
 		},
 	}
 )
 
+func locksToCookbooks(locks map[string]chef.CookbookLock) []reporting.NodeCookbook {
+	var cbs = make([]reporting.NodeCookbook, 0)
+
+	for name, lock := range locks {
+		cbs = append(cbs, reporting.NodeCookbook{Name: name, Version: lock.Identifier})
+	}
+	return cbs
+}
+
 func init() {
 	addInfraFlagsToCommand(captureCmd)
 }
-func requestCookbookPathFullMessage(repoPath string, cookbooks []reporting.NodeCookbook) string {
-	var input, path string
 
-	// We issue a long instruction and ask to press Enter to continue.
-	// This is out of fear that a long cookbook list would scroll the
-	// instructions off the screen.
-	fmt.Printf(CookbookCaptureGatherSourcesTxt, repoPath)
-	fmt.Scanf("%s\n", &input) // discarded
+func requestCookbookPath(cookbooks []reporting.NodeCookbook) string {
+
+	userPath := promptUser(fmt.Sprintf(CookbookCaptureRequestCookbookPathTxt, formatCookbooks(cookbooks)))
+	for len(userPath) > 0 {
+		absPath, _ := filepath.Abs(userPath)
+		if _, err := os.Stat(absPath); err == nil {
+			break
+		}
+		userPath = promptUser(fmt.Sprintf("'%s' is not a valid path. Cookbook Location [none]: ", userPath))
+	}
+	return userPath
+}
 
-	fmt.Printf(CookbookCaptureRequestCookbookPathTxt, formatCookbooks(cookbooks))
-	fmt.Scanf("%s\n", &path)
-	return path
+func requestGatherSources(repoPath string) {
+	fmt.Printf(CookbookCaptureGatherSourcesTxt, repoPath)
+	promptUser(PressEnterToContinueTxt)
 }
 
-func requestCookbookPath(cookbooks []reporting.NodeCookbook) string {
-	fmt.Printf(CookbookCaptureRequestCookbookPathTxt, formatCookbooks(cookbooks))
-	var path string
-	fmt.Scanf("%s\n", &path)
-	return path
+func promptUser(msg string) string {
+	fmt.Print(msg)
+	var answer string
+	fmt.Scanf("%s\n", &answer)
+	fmt.Println("") // Blank linke after taking input separates any response we may write
+	return answer
 }
 
 func formatCookbooks(cookbooks []reporting.NodeCookbook) string {
 	var cbNames []string
 	for _, cb := range cookbooks {
+		// TODO - do we want the hash shown for cookbook artifacts? Does it matter?
```

Comment:
I don't think so. It looks like this function is just used to convey information to the user, and the cookbook name is the most important part of that. Does the cookbook folder on disk include the SHA in its name?

Reply:
Because we store cookbooks in either a `cookbooks` or `cookbooks_artifacts` folder I think it would make more sense to display that path here (EG `- cookbooks/my_cookbook` or `- cookbook_artifacts/my_cookbook-0.1.0`) but I think that can be a future improvement

Reply:
The cookbook directory on disk will include the SHA because that's how chef-zero needs it,  like `cookbook_artifacts/CBNAME-SHA`; in the same way that cookbooks are stored `cookbooks/CBNAME-VERSION`